automation/github-source-tarball-signature.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright the KubeVirt Authors.
+#
+#
+
+function update_github_source_tarball_signature() {
+    local src_tarball_file
+    src_tarball_file="${DOCKER_TAG}.tar.gz"
+    src_tarball_signature_file="${src_tarball_file}.asc"
+
+    if ! gh release download "$DOCKER_TAG" --repo "$GITHUB_REPOSITORY" --pattern="${src_tarball_signature_file}" --clobber --output "/tmp/${src_tarball_signature_file}"; then
```

Comment:
Hey instead of dumping a bunch of files into /tmp, could we do something like:

TMPDIR=$(mktemp -d)
..... --output "${TMPDIR}/${src_tarball_signature_file}"

point being it keeps each run in a separate isolated directory that cannot be re-used. This reduces weird collision issues between multiple runs to zero.

Added bonus, cleanup is a breeze.

Reply:
Actually. my suggestion stands but don't name the variable TMPDIR. that name is generally honored by various commands to be "the directory to use instead of /tmp".

Reply:
@stu-gott good point with the subdirectory creation - would it be ok if I do that in a follow up PR, so that we get this merged as is?

1) there's only one few specific files that are being output
2) this will rather run in a pod, so there shouldn't be an issue with tmp collisions, only when being run locally by a user

WDYT?